test/integration_test/pure/types.go

```diff
@@ -0,0 +1,30 @@
+package pure
```

Comment:
Could we just put all the types in one folder instead of having types for every type of test? Otherwise we will have a lot of folders like these.
May be we can just create a folder called types and have different filenames for each of them - 
```
test/
└── integration_test
    └── types
       ├── pure.go
       └── testcase.go
```

Reply:
IMO we need different packages to serve different purposes that's why folders are created. we can have public constants and functions in different packages, so when calling them it's more clear. if we put everything into one package, I feel it's a little bit hard to see what's in the package since there are a lot, and I feel it's a little strange to call `types.TestCase`.. what do you think? but I agree that way from structure wise it's easier to maintain

Reply:
I'm not sure which approach is better, can I get more inputs? @haibinxie @Pure-AdamuKaapan

Reply:
The `types` is just a suggestion, you can name it to whatever makes sense. Although `types.TestCase` does not sound that weird to me. May be because kubernetes has some `types.*` objects.

But anyway, I just wanted to group all the testing related structs in one place instead of having those structs all over the place in different folders. That would create a lot of folders.

Reply:
ok makes sense, will update the PR.